### ISSUE TITLE
Update to Gradle 5.0

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1350]
Update to Gradle 5.0 https://docs.gradle.org/5.0/release-notes.html 
- [x] CQ (https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18369) 
- [x] our own builds
- [x] wizard
- [x] buildship? (oomph) https://projects.eclipse.org/projects/tools.buildship/downloads
- [x] have a look for deprecations in gradle 6.0 and open a new issue if there are any

**Comment by cdietrich** [comment]
```
/data/xtext-master/git/xtext-web$ ./gradlew clean build -PuseJenkinsSnapshots=true --warning-mode all

> Configure project :
The osgi plugin has been deprecated. This is scheduled to be removed in Gradle 6.0. Consider using the biz.aQute.bnd plugin instead.
        at manifest_gen_6vfzhhhsie8v8m1761v0603i0.run(/data/xtext-master/git/xtext-web/gradle/manifest-gen.gradle:6)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)

```

**Comment by cdietrich** [comment]
JavaSourceLanguagePerformanceTest fails with Gradle 5

```
org.junit.runners.model.TestTimedOutException: test timed out after 30000 milliseconds
	at java.lang.System.identityHashCode(Native Method)
	at org.eclipse.jdt.internal.compiler.lookup.TypeSystem$HashedParameterizedTypes$PTBKey.hash(TypeSystem.java:114)
	at org.eclipse.jdt.internal.compiler.lookup.TypeSystem$HashedParameterizedTypes$PTBKey.hashCode(TypeSystem.java:122)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at org.eclipse.jdt.internal.compiler.lookup.TypeSystem$HashedParameterizedTypes.get(TypeSystem.java:148)
	at org.eclipse.jdt.internal.compiler.lookup.TypeSystem.getParameterizedType(TypeSystem.java:316)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.createParameterizedType(LookupEnvironment.java:1308)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getTypeFromTypeSignature(LookupEnvironment.java:1932)
	at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.createMethod(BinaryTypeBinding.java:852)
	at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.createMethods(BinaryTypeBinding.java:969)
	at org.eclipse.jdt.internal.compiler.lookup.BinaryTypeBinding.cachePartsFrom(BinaryTypeBinding.java:531)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.createBinaryTypeFrom(LookupEnvironment.java:1022)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.createBinaryTypeFrom(LookupEnvironment.java:1003)
	at org.eclipse.jdt.internal.compiler.Compiler.accept(Compiler.java:303)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.askForType(LookupEnvironment.java:317)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getType(LookupEnvironment.java:1626)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getResolvedType(LookupEnvironment.java:1548)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.getResolvedJavaBaseType(LookupEnvironment.java:1560)
	at org.eclipse.jdt.internal.compiler.lookup.Scope.getJavaLangObject(Scope.java:2824)
	at org.eclipse.jdt.internal.compiler.lookup.ClassScope.connectSuperclass(ClassScope.java:954)
	at org.eclipse.jdt.internal.compiler.lookup.ClassScope.connectTypeHierarchy(ClassScope.java:1119)
	at org.eclipse.jdt.internal.compiler.lookup.CompilationUnitScope.connectTypeHierarchy(CompilationUnitScope.java:359)
	at org.eclipse.jdt.internal.compiler.lookup.LookupEnvironment.completeTypeBindings(LookupEnvironment.java:500)
	at org.eclipse.jdt.internal.compiler.Compiler.internalBeginToCompile(Compiler.java:861)
	at org.eclipse.jdt.internal.compiler.Compiler.beginToCompile(Compiler.java:387)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:437)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:419)
	at org.eclipse.xtext.java.resource.JavaDerivedStateComputer.installFull(JavaDerivedStateComputer.java:260)
	at org.eclipse.xtext.java.resource.JavaResource.lambda$installFull$1(JavaResource.java:195)
	at org.eclipse.xtext.java.resource.JavaResource$$Lambda$36/1169087811.apply(Unknown Source)
	at org.eclipse.xtext.java.resource.JavaResource.initializing(JavaResource.java:205)
	at org.eclipse.xtext.java.resource.JavaResource.installFull(JavaResource.java:199)
	at org.eclipse.xtext.java.resource.JavaResource.getContents(JavaResource.java:168)
	at org.eclipse.xtext.java.tests.JavaSourceLanguagePerformanceTest.doTestPerformance(JavaSourceLanguagePerformanceTest.java:94)
	at org.eclipse.xtext.java.tests.JavaSourceLanguagePerformanceTest.testPerf51200(JavaSourceLanguagePerformanceTest.java:152)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by cdietrich** [comment]
@oehme were there any changes in gradle 5 how the test vm allocates memory
if i set xmx explicitely

```
test {
	maxHeapSize = "2G"
}
```
test seems ok

in 4.10.2 test worker looks like

![bildschirmfoto vom 2018-12-04 17-35-42](https://user-images.githubusercontent.com/238736/49457419-1f577280-f7eb-11e8-82e8-df3f3d34bdfd.png)

**Comment by oehme** [comment]
Yes, this change was intentional, see the 5.0 [migration guide](https://docs.gradle.org/5.0/userguide/upgrading_version_4.html). Before the memory was using the JVM default, which is a crazy 1/4th of total RAM.

**Comment by cdietrich** [comment]
Fixed in 2.17
